cmd/tdex/depositfee.go

```diff
@@ -27,17 +54,256 @@ func depositFeeAction(ctx *cli.Context) error {
 	}
 	defer cleanup()
 
-	numOfAddresses := ctx.Int64("num_of_addresses")
+	net, baseAssetKey := getNetworkAndBaseAssetKey(ctx.String("network"))
+	fragmentationDisabled := ctx.Bool("no-fragment")
+
+	if fragmentationDisabled {
+		numOfAddresses := int64(1)
+		if ctx.Int64("num_of_addresses") > 1 {
+			numOfAddresses = ctx.Int64("num_of_addresses")
+		}
+		resp, err := client.DepositFeeAccount(
+			context.Background(), &pboperator.DepositFeeAccountRequest{
+				NumOfAddresses: numOfAddresses,
+			},
+		)
+		if err != nil {
+			return err
+		}
+
+		printRespJSON(resp)
+
+		return nil
+	}
+
+	randomWallet, err := trade.NewRandomWallet(&net)
+	if err != nil {
+		return err
+	}
+
+	log.Warnf("fund address: %v", randomWallet.Address())
+	explorerSvc, err := config.GetExplorer()
+	if err != nil {
+		log.WithError(err).Panic("error while setting up explorer service")
+	}
+	baseAssetPair, unspents := findBaseAssetsUnspents(
+		randomWallet,
+		explorerSvc,
+		baseAssetKey,
+	)
+
+	log.Info("calculating fragments ...")
+	baseFragments := fragmentFeeUnspents(
+		baseAssetPair.BaseValue,
+		MinFee,
+		MaxNumOfOutputs,
+	)
+	log.Infof("base fragments: %v", baseFragments)
+
+	addresses, err := fetchFeeAccountAddresses(
+		len(baseFragments),
+		client,
+	)
+	if err != nil {
+		log.Error(err)
+	}
+
+	feeAmount := wallet.EstimateTxSize(
+		len(unspents),
+		len(baseFragments),
+		false,
+		MinMilliSatPerByte,
+	)
+
+	outputs := createOutputsForDepositFeeTransaction(
+		baseFragments,
+		feeAmount,
+		addresses,
+		baseAssetKey,
+	)
+
+	txHex, err := craftTransaction(
+		randomWallet,
+		unspents,
+		outputs,
+		feeAmount,
+		net,
+		baseAssetKey,
+		[][]byte{randomWallet.BlindingKey()},
+	)
+	if err != nil {
+		log.Error(err)
+	}
+
+	var txID string
+retry:
+	for {
+		resp, err := explorerSvc.BroadcastTransaction(txHex)
+		if err != nil {
+			log.Warn(err)
+			log.Info("transaction broadcast retry")
+			continue retry
+		}
+		log.Info(resp)
+		txID = resp
+		break retry
+	}
```

Comment:
why retrying? It's unlikely that retrying to broadcast a tx makes it accepted in mempool if the first time it didn't. If the tx gets rejected, it's because it's invalid, ie. break the Elements protocol, and retrying won't make any difference. Am I wrong?